### PR TITLE
fix(service-portal): update schema and models for inna client

### PR DESCRIPTION
**Pull request by Valur**
## What

Updated schema and models for inna client

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/education/src/lib/graphql/inna/periods.model.ts [hunk 1]
```diff
@@ -20,8 +20,8 @@ export class CoursesModel {
   @Field(() => String, { nullable: true })
   units?: number
 
-  @Field(() => String, { nullable: true })
-  stage?: string
+  @Field(() => Number, { nullable: true })
+  stage?: number | null
```
Comment (thorkellmani) [inline libs/api/domains/education/src/lib/graphql/inna/periods.model.ts:24]: Isn't unnecessary to have it nullable and also null?

File: libs/clients/inna/src/lib/inna.config.ts [hunk 2]
```diff
@@ -2,17 +2,16 @@ import { defineConfig } from '@island.is/nest/config'
...
   name: 'InnaClientConfig',
   schema,
   load: (env) => ({
-    fetch: {
-      timeout: 30000,
-    },
+    xRoadServicePath: undefined /* env.required(
+      'XROAD_INNA_PATH',
+      'IS-DEV/GOV/',
+    )*/,
```
Comment (thorkellmani) [inline libs/clients/inna/src/lib/inna.config.ts:15]: Inna yet to implement xroad?

**Review by thorkellmani — APPROVED** [review]
LGTM

File: libs/clients/inna/src/lib/inna.config.ts (hunk 2 above)
Comment (Valur) [reply libs/clients/inna/src/lib/inna.config.ts:15]: No, the xroad connection is disabled for now, the client is working on implementing it

File: libs/api/domains/education/src/lib/graphql/inna/periods.model.ts (hunk 1 above)
Comment (Valur) [reply libs/api/domains/education/src/lib/graphql/inna/periods.model.ts]: Yea, fixed

File: libs/api/domains/education/src/lib/graphql/inna.resolver.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ import { DiplomaModel } from './inna/diplomas.model'
 @FeatureFlag(Features.servicePortalSecondaryEducationPages)
 @Audit({ namespace: '@island.is/api/education-inna' })
 export class InnaResolver {
-  constructor(private innaService: InnaService) {}
+  constructor(private innaService: InnaClientService) {}
```
Comment (kksteini) [inline libs/api/domains/education/src/lib/graphql/inna.resolver.ts:30]: Seeing some warnings from Github about 
`Type 'InlineResponse2001 | null' is not assignable to type 'PeriodsModel | null'.`

Does this change trigger this or is it something incidental that can be ignored?

File: libs/clients/inna/src/lib/inna.provider.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          logErrorResponseBody: true,
+          /*autoAuth: idsClientConfig.isConfigured
+            ? {
+                mode: 'tokenExchange',
+                issuer: idsClientConfig.issuer,
+                clientId: idsClientConfig.clientId,
+                clientSecret: idsClientConfig.clientSecret,
+                scope: ['@mms.is/inna'],
+              }
+            : undefined,*/
```
Comment (kksteini) [inline libs/clients/inna/src/lib/inna.provider.ts:34]: To be implemented?

**Review by kksteini — APPROVED** [review]
LGTM, some minor thoughts.

File: libs/clients/inna/src/lib/inna.provider.ts (hunk 4 above)
Comment (Valur) [reply libs/clients/inna/src/lib/inna.provider.ts:34]: Yes, the api is being implemented in x-road

File: libs/api/domains/education/src/lib/graphql/inna.resolver.ts (hunk 3 above)
Comment (Valur) [reply libs/api/domains/education/src/lib/graphql/inna.resolver.ts:30]: Yea I need to fix it before merging, missing null value in model

**Review by lodmfjord — APPROVED** [review]
lgtm

**Pull request merged**